target/linux/ath79/generic/base-files/etc/board.d/02_network

```diff
@@ -160,6 +160,11 @@ ath79_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"0@eth0" "1:wan" "2:lan:3" "3:lan:2"
 		;;
+	engenius,ens202ext)
+		ucidef_set_interface_wan "eth1"
+		ucidef_add_switch "switch0" \
+			"0@eth0" "1:lan"
+		;;
```

Comment:
Is there a specific reason for the switch setup? Because, I'd just have dropped this and went for the default case, where wan will be eth1 and lan will be eth0?

Reply:
at one point in testing I had

`ucidef_set_interface_lan "eth0 eth1"`

and only eth1 was working.

Does phy-mode have anything to do with this? for eth0 it is "gmii" by default. Would it function maybe if it is "mii"?

BTW see my old commit in ar71xx. I had a very hard time with the built-in switch on this board in the past.
#1845

Reply:
Had no luck in any other configuration of gmac0

Reply:
But still `ucidef_set_interfaces_lan_wan "eth0" "eth1"` (default case) should work?

Reply:
> But still `ucidef_set_interfaces_lan_wan "eth0" "eth1"` (default case) should work?

It's better to have the switch set up properly. It'll be painful when we want to switch to a dsa driver in the future if we omit the switch setup now.

Reply:
eth0 does not function if I do `ucidef_set_interfaces_lan_wan "eth0" "eth1"`

I don't know why